platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java

```diff
@@ -221,26 +188,23 @@ public boolean equals(final Object obj) {
             return false;
         }
 
-        return digestType.id() == that.digestType.id() && Arrays.equals(value, that.value);
-    }
-
-    /**
-     * Check if the bytes of this hash are equal to the bytes supplied.
-     *
-     * @param bytes
-     * 		the bytes to compare
-     * @return true if the bytes are equal, false otherwise
-     */
-    public boolean equalBytes(@Nullable final Bytes bytes) {
-        return bytes != null && value.length == bytes.length() && bytes.contains(0, value);
+        return digestType.id() == that.digestType.id() && Objects.equals(bytes, that.bytes);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public int hashCode() {
-        return ((value[0] << 24) + (value[1] << 16) + (value[2] << 8) + (value[3]));
+        // A few notes about this implementation:
+        // - it is very important that this method has very low overhead, some usages are very performance sensitive
+        // - since this is a cryptographic hash, it is safe to use the first 4 bytes of the hash as the hash code.
+        //   it fulfills the requirements of hashCode, it is consistent with equals, and it is very unlikely to conflict
+        //   with other hash codes. by using the first 4 bytes, we can avoid unnecessary overhead
+        // - it does not make a difference if the bytes are big endian or little endian, as long as they are consistent.
+        //   this is why we don't specify the order when calling getInt(), we use whatever order is the default to avoid
+        //   the overhead of reordering the bytes
```

Comment:
I'd prefer if we use the getInt(long, ByteOrder) still. But I'm okay as long as we rely on the default byte order consciously as this comment suggests.

Reply:
maybe I didn't phrase it well, what I meant was consistent during runtime. if it changes after a restart, that is fine.

Reply:
> I'd prefer if we use the getInt(long, ByteOrder) 

`hashCode()` returns an int, so there is no benefit to getting a long. The higher order bits will just be chopped off when it is cast to an int.

Reply:
> This appears to place a requirement that the default never change, and remain identical across system architectures.

I don't think we should make this a contract of this class, even if we happen to never change its implementation. Any time we are relying on `hashCode()` to return the same value value between JVM runs should be a considered a code smell. I recall there was an issue with virtual maps a while back where it broke because it assumed `hashCode()` would never change. This is a super dangerous assumption I think.

Reply:
Definitely the wording is a bit confusing here.  The following _might_ be clearer, though it's not easy to express in a short comment. The comment confusion is not critical, and can be done separately.
```suggestion
        // - it does not make a difference if the bytes are big endian or little endian, as long as the order
        //   is reasonably stable. This is why we don't specify the byte order here; we can accept the current
        //   default byte order and avoid the overhead of reordering the bytes
```